tests/conftest.py

```diff
@@ -206,7 +206,7 @@ def fanouthosts(ansible_adhoc, conn_graph_facts, creds):
     Shortcut fixture for getting Fanout hosts
     """
 
-    dev_conn     = conn_graph_facts['device_conn']
+    dev_conn     = conn_graph_facts['device_conn'] if 'device_conn' in conn_graph_facts else {}
```

Comment:
The device_conn doesn't exists when using virtual testbed. Add this code to avoid test broken